include/exadg/grid/grid.cpp

```diff
@@ -160,10 +164,51 @@ Grid<dim>::create_triangulation(
     {
       // we need to generate the coarse triangulations explicitly
 
-      // TODO
-      // construct all the coarser triangulations
+      // resize the empty coarse triangulations vector
+      coarse_triangulations = std::vector<std::shared_ptr<dealii::Triangulation<dim> const>>(
+        triangulation->n_global_levels());
+
+      // we start with the finest level
+      unsigned int level = triangulation->n_global_levels() - 1;
+
+      // lambda function for creating and placing the coarse triangulations
+      auto const lambda_create_level_triangulation = [&](unsigned int              refine_global,
+                                                         std::vector<unsigned int> refine_local) {
+        auto level_tria = std::make_shared<dealii::parallel::fullydistributed::Triangulation<dim>>(
+          triangulation->get_communicator());
+
+        do_create_triangulation(
+          level_tria, data, create_coarse_triangulation, refine_global, refine_local);
+
+        coarse_triangulations[level] = level_tria;
+
+        level--;
+      };
 
-      AssertThrow(false, ExcNotImplemented());
+      // start with the finest level
+      std::vector<unsigned int> refine_local = vector_local_refinements;
+
+      // undo global refinements
+      for(int it = global_refinements; it >= 0; --it)
```

Comment:
a last suggestion for renaming would be to use `refine_global` here instead of `it`.

Reply:
done!